File: DSA/Ayush_Mishra_590014844/Day_10_Question_1.py
def compare_strings(s1, s2):
    if s1 == s2:
        print("Equal")
    else:
        print("Not equal")

    if s1.lower() == s2.lower():
        print("Equal ignoring case")
    else:
        print("Not equal ignoring case")

    if s1 == s2:
        print("Strings are identical")
    elif s1 < s2:
        print(f'"{s1}" comes before "{s2}"')
    else:
        print(f'"{s2}" comes before "{s1}"')

File: DSA/Ayush_Mishra_590014844/Day_10_Question_1.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>
void compareStrings(char *s1, char *s2) {
    if (strcmp(s1, s2) == 0)
        printf("Equal\n");
    else
        printf("Not equal\n");

    // Case-insensitive comparison
    if (strcasecmp(s1, s2) == 0)
        printf("Equal ignoring case\n");
    else
        printf("Not equal ignoring case\n");

    int cmp = strcmp(s1, s2);
    if (cmp == 0)
        printf("Strings are identical\n");
    else if (cmp < 0)
        printf("\"%s\" comes before \"%s\"\n", s1, s2);
    else
        printf("\"%s\" comes before \"%s\"\n", s2, s1);
}
